src/unzip/mod.rs

```diff
@@ -359,12 +366,22 @@ fn extract_file_by_index<'a, T: Read + Seek + 'a>(
     get_ziparchive_clone: impl Fn() -> ZipArchive<T> + Sync,
     i: usize,
     output_directory: &Option<PathBuf>,
+    password: &Option<String>,
     progress_reporter: &dyn UnzipProgressReporter,
     directory_creator: &DirectoryCreator,
 ) -> Result<(), anyhow::Error> {
     let myzip: &mut zip::ZipArchive<T> = &mut get_ziparchive_clone();
-    let file = myzip.by_index(i)?;
-    extract_file(file, output_directory, progress_reporter, directory_creator)
+    let file: ZipFile;
```

Comment:
Nit: unless there's some reason this is impossible, please could you:

```rust
let file = match password {
   None => myzip.by_index(i)?,
   Some(password) => myzip.by_index_decrypt(i, string.as_bytes())??,
};
extract_file(file, ...);
```

just to reduce duplication.

Reply:
Thanks for this feedback. I've made a change. This is much nicer.